shopfloor/services/zone_picking.py

```diff
@@ -869,10 +869,11 @@ def _set_destination_location(self, move_line, quantity, confirmation, location)
         # if `confirmation is True
         # Ask confirmation to the user if the scanned location is not in the
         # expected ones but is valid (in picking type's default destination)
-        if not self.is_dest_location_valid(move_line.move_id, location):
+        message = self._validate_destination(location, move_line.move_id)
+        if message:
```

Comment:
looks like a handler to me :wink:

Reply:
Everything looks like a nail when you have a hammer.

Reply:
aha